director/origin_api.go

```diff
@@ -271,18 +274,49 @@ func VerifyDirectorTestReportToken(strToken string) (bool, error) {
 	return false, nil
 }
 
-func GetRegistryIssuerURL(prefix string) (string, error) {
-	namespace_url_string := param.Federation_RegistryUrl.GetString()
-	if namespace_url_string == "" {
-		return "", errors.New("Namespace URL is not set")
+// For a given prefix, get the url of the issuer/public key from the registry
+// This works by looking up the namespace-configuration json for the namespace
+// and grabbing the value corresponding to the "jwks_uri" key.
+func GetNSIssuerURL(prefix string) (string, error) {
+	if prefix == "" || !strings.HasPrefix(prefix, "/") {
+		return "", errors.New(fmt.Sprintf("the prefix \"%s\" is invalid", prefix))
 	}
-	namespace_url, err := url.Parse(namespace_url_string)
+	registryUrlStr := param.Federation_RegistryUrl.GetString()
+	if registryUrlStr == "" {
+		return "", errors.New("federation registry URL is not set and was not discovered")
+	}
+	registryUrl, err := url.Parse(registryUrlStr)
 	if err != nil {
 		return "", err
 	}
-	namespace_url.Path, err = url.JoinPath(namespace_url.Path, "api", "v1.0", "registry", prefix, ".well-known", "issuer.jwks")
+
+	registryUrl.Path, err = url.JoinPath(registryUrl.Path, "api", "v1.0", "registry", prefix, ".well-known", "openid-configuration")
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "failed to construct namespace-configuration lookup URL for prefix %s", prefix)
+	}
+
+	// Get/parse the namespace-configuration JSON to lookup key location
+	client := &http.Client{Transport: config.GetTransport()}
+	originConfig, err := client.Get(registryUrl.String())
```

Comment:
Do we want to consider backward compatibility and fallback option here? At least for OSDF it doesn't seem to have such an endpoint